src/Microsoft.ML.Data/Transforms/ConversionsExtensionsCatalog.cs

```diff
@@ -92,24 +92,24 @@ public static KeyToValueMappingEstimator MapKeyToValue(this TransformsCatalog.Co
         /// <param name="outputColumn">The name of the output column.</param>
         /// <param name="bag">Whether bagging is used for the conversion. </param>
         public static KeyToVectorMappingEstimator MapKeyToVector(this TransformsCatalog.ConversionTransforms catalog,
-            string inputColumn, string outputColumn = null, bool bag = KeyToVectorMappingEstimator.Defaults.Bag)
-            => new KeyToVectorMappingEstimator(CatalogUtils.GetEnvironment(catalog), inputColumn, outputColumn, bag);
+            string outputColumn, string inputColumn, bool bag = KeyToVectorMappingEstimator.Defaults.Bag)
```

Comment:
>inputColumn [](start = 40, length = 11)

I'm not a Tom, and I can't speak from his name, but I don't remember him objecting on idea of one of column be optional, and I thinks it's perfectly fine to have second column nullable and replace it with first one.

Reply:
Yes, in fact that was our prior behavior.

---
In reply to: [246971850](https://github.com/dotnet/machinelearning/pull/2118#discussion_r246971850) [](ancestors = 246971850)